### ISSUE TITLE
Open Lidar toolbox problem running LidarPointDensity

### ISSUE THREAD

**Issue by ericawynter** [stefaneichert/OpenLidarToolbox#2]
Hi Stefan,

I am haven a problem setting up the Open Lidar Toolbox.  I have copied the whole of the script below.  The problem appears to be specific to the whitebox tool setup and the LidarPointSetup tool.  I am not sure if this has to do with the bruy plugin, I did get a zipped copy of that plugin from the whitebox forum.  Apologies for the bug but I am fairly desperate for some assist as I would really like to get this setup and working before the TRAIL meeting in April.  I have sent a query through to Whitebox as well but am hoping you can assist.

QGIS version: 3.24.0-Tisler
QGIS code revision: 6b44a42058
Qt version: 5.15.2
Python version: 3.9.5
GDAL version: 3.4.1
GEOS version: 3.10.2-CAPI-1.16.0
PROJ version: Rel. 8.2.1, January 1st, 2022
PDAL version: 2.3.0 (git-version: 9f35b7)
Algorithm started at: 2022-03-07T10:04:17
Algorithm '0NE (One-Step-Processing)' starting…
Input parameters:
{ 'CRS' : QgsCoordinateReferenceSystem('EPSG:4326'), 'GPD' : True, 'InputFilelaslaz' : 'C:\\Users\\ewynter\\OneDrive\\_Masters\\Aerial photos\\Full _001.las', 'LAS' : 'TEMPORARY_OUTPUT', 'LVD' : True, 'LowNoise' : False, 'SetCellSize' : 0.5, 'VisualisationCM' : True, 'VisualisationDFM' : True, 'VisualisationDfME' : True, 'VisualisationHS' : True, 'VisualisationOPN' : True, 'VisualisationSVF' : True, 'VisualisationVAT' : True, 'classLas' : False, 'prefix' : '' }

LAStools command line
C:\LAStools\bin\lasground -v -i "C:\Users\ewynter\OneDrive\_Masters\Aerial photos\Full _001.las" -ignore_class 7 -city -ultra_fine -o "C:/Users/ewynter/AppData/Local/Temp/processing_DwZsjm/66af3ed653dc4e37a8a87a4b29c097dc/lasground1.laz"
LAStools console output
The system cannot find the path specified. 
Results: {'': None}
LAStools command line
C:\LAStools\bin\lasheight -v -i "C:/Users/ewynter/AppData/Local/Temp/processing_DwZsjm/66af3ed653dc4e37a8a87a4b29c097dc/lasground1.laz" -o "C:/Users/ewynter/AppData/Local/Temp/processing_DwZsjm/4ff564fa40d54915a9d6dc7746a2f48e/lasheight.laz" -ignore_class 0 1 3 4 5 6 7 8 9 10 11 12
LAStools console output
The system cannot find the path specified. 
Results: {'': None}
LAStools command line
C:\LAStools\bin\lasclassify -v -i "C:/Users/ewynter/AppData/Local/Temp/processing_DwZsjm/4ff564fa40d54915a9d6dc7746a2f48e/lasheight.laz" -o "C:/Users/ewynter/AppData/Local/Temp/processing_DwZsjm/03dfebf0de9546858e4345ea813aba06/lasclassify.laz" -small_trees
LAStools console output
The system cannot find the path specified. 
Results: {'': None}
LAStools command line
C:\LAStools\bin\lasground -v -i "C:/Users/ewynter/AppData/Local/Temp/processing_DwZsjm/03dfebf0de9546858e4345ea813aba06/lasclassify.laz" -wilderness -ultra_fine -o "C:/Users/ewynter/AppData/Local/Temp/processing_DwZsjm/1f74907b157d4577bec136f01ecc0825/lasground2.laz" -ignore_class 6 7
LAStools console output
The system cannot find the path specified. 
Results: {'': None}
LAStools command line
C:\LAStools\bin\lasheight -v -i "C:/Users/ewynter/AppData/Local/Temp/processing_DwZsjm/1f74907b157d4577bec136f01ecc0825/lasground2.laz" -ignore_class 6 -ignore_class 7 -classify_below -0.25 7 -classify_between -0.2 0.2 2 -classify_between 0.5 2.0 3 -classify_above 2.0 5 -o "C:/Users/ewynter/AppData/Local/Temp/processing_DwZsjm/ea48a3761f724576b44186bf5984b2ba/lasheightCl.las"
LAStools console output
The system cannot find the path specified. 
Results: {'': None}
Results: {'classifiedLAZ': 'C:/Users/ewynter/AppData/Local/Temp/processing_DwZsjm/ea48a3761f724576b44186bf5984b2ba/lasheightCl.las'}
WhiteboxTools command:
"C:/OSGEO4W64/WBT/whitebox_tools.exe" -v --run=LidarPointDensity --input="C:\Users\ewynter\AppData\Local\Temp\processing_DwZsjm\ea48a3761f724576b44186bf5984b2ba\lasheightCl.las" --returns="all" --resolution=2.0 --radius=10.0 --exclude_cls="0,1,3,4,5,7,8" --output="C:\Users\ewynter\AppData\Local\Temp\processing_DwZsjm\7192d8302a1f4f36b79ccd2183d7eb31\gpd.tif"
WhiteboxTools output:
********************************
* Welcome to LidarPointDensity *
* Powered by WhiteboxTools *
* [www.whiteboxgeo.com](http://www.whiteboxgeo.com/) *
********************************
thread 'main' panicked at 'The system cannot find the file specified. (os error 2)', whitebox-tools-app\src\main.rs:72:21
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Process returned error code 101
Results: {'output': 'C:/Users/ewynter/AppData/Local/Temp/processing_DwZsjm/7192d8302a1f4f36b79ccd2183d7eb31/gpd.tif'}
Unable to execute algorithm
Could not load source layer for input: C:/Users/ewynter/AppData/Local/Temp/processing_DwZsjm/7192d8302a1f4f36b79ccd2183d7eb31/gpd.tif not found
Unable to execute algorithm
Could not load source layer for input: C:/Users/ewynter/AppData/Local/Temp/processing_DwZsjm/7192d8302a1f4f36b79ccd2183d7eb31/gpd.tif not found
Execution failed after 0.31 seconds

Loading resulting layers
Algorithm '0NE (One-Step-Processing)' finished

Regards

Erica Wynter

**Comment by stefaneichert** [comment]
Hi Erica,
Thank you very much for reporting this. The reason seems to be somehow connected to the slashes and backslashes that the input resp. outputfiles use for their path. Wich operating system are you using? I assume Windows?
I am looking into this and we are working on a solution until April.
best, Stefan

**Comment by stefaneichert** [comment]
Hi Erica! Have you installed the whitebox and lastools executables on your machine too? Or did you only install the Qgis plugins alone?

**Comment by ericawynter** [comment]
Hi Stefan,  managed to install the whiteboard tools, it was a bit of a 
workaround as the bruy plug -in was not available.

Apologies for tardiness in replying, we have had a week of power supply 
interruptions and its been a bit crazy. I will forward you the emails I 
sent Benjamin in the morning. I did have a reply from whitebox but it was 
not helpful unfortunately.

Regards

Erica

On 10 March 2022 14:54:17 Stefan Eichert ***@***.***> wrote:
>
> Hi Erica! Have you installed the whitebox and lastools executables on your 
> machine too? Or did you only install the Qgis plugins alone?
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> Triage notifications on the go with GitHub Mobile for iOS or Android.
> You are receiving this because you authored the thread.Message ID: 
> ***@***.***>


Sent with Aqua Mail for Android
https://www.aqua-mail.com

**Comment by stefaneichert** [comment]
So it is working now?

**Comment by ericawynter** [comment]
Hi Stefan,

 

I have copied the error below, it is still not working.  I followed the install instructions as per your OpenLidarToolbox.  The only problem I ran into was the BRUY plugin.  I used a zip copy of that plugin and installed it separately.  I just ran the program again and got the same error.  I have tried this on two different pc’s and the same problem persists.

 

Any suggestions would be appreciated.

 

QGIS version: 3.22.3-Białowieża

QGIS code revision: 1628765ec7

Qt version: 5.15.2

Python version: 3.9.5

GDAL version: 3.4.1

GEOS version: 3.10.0-CAPI-1.16.0

PROJ version: Rel. 8.2.1, January 1st, 2022

PDAL version: 2.3.0 (git-version: 9f35b7)

Algorithm started at: 2022-03-11T12:09:43

Algorithm '0NE (One-Step-Processing)' starting…

Input parameters:

{ 'CRS' : QgsCoordinateReferenceSystem('EPSG:32735'), 'GPD' : True, 'InputFilelaslaz' : 'H:\\PHDATA\\PHLaz\\Full _010.laz', 'LAS' : 'TEMPORARY_OUTPUT', 'LVD' : True, 'LowNoise' : False, 'SetCellSize' : 0.5, 'VisualisationCM' : True, 'VisualisationDFM' : True, 'VisualisationDfME' : True, 'VisualisationHS' : True, 'VisualisationOPN' : True, 'VisualisationSVF' : True, 'VisualisationVAT' : True, 'classLas' : False, 'prefix' : '' }

 

LAStools command line

C:\LAStools\bin\lasground -v -i "H:\PHDATA\PHLaz\Full _010.laz" -ignore_class 7 -city -ultra_fine -o "C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_avyDJN/efa403277e1b45649ae854f195dc39d8/lasground1.laz"

LAStools console output

The system cannot find the path specified. 

Results: {'': None}

LAStools command line

C:\LAStools\bin\lasheight -v -i "C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_avyDJN/efa403277e1b45649ae854f195dc39d8/lasground1.laz" -o "C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_avyDJN/c78c2b5ad815494fab5a243b1c2198b4/lasheight.laz" -ignore_class 0 1 3 4 5 6 7 8 9 10 11 12

LAStools console output

The system cannot find the path specified. 

Results: {'': None}

LAStools command line

C:\LAStools\bin\lasclassify -v -i "C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_avyDJN/c78c2b5ad815494fab5a243b1c2198b4/lasheight.laz" -o "C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_avyDJN/e809bd0c13e746ae81619bb2d3b9fc88/lasclassify.laz" -small_trees

LAStools console output

The system cannot find the path specified. 

Results: {'': None}

LAStools command line

C:\LAStools\bin\lasground -v -i "C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_avyDJN/e809bd0c13e746ae81619bb2d3b9fc88/lasclassify.laz" -wilderness -ultra_fine -o "C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_avyDJN/7fc7ab7160c94543acf75b4b882e571f/lasground2.laz" -ignore_class 6 7

LAStools console output

The system cannot find the path specified. 

Results: {'': None}

LAStools command line

C:\LAStools\bin\lasheight -v -i "C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_avyDJN/7fc7ab7160c94543acf75b4b882e571f/lasground2.laz" -ignore_class 6 -ignore_class 7 -classify_below -0.25 7 -classify_between -0.2 0.2 2 -classify_between 0.5 2.0 3 -classify_above 2.0 5 -o "C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_avyDJN/64a1ce455e2f484c985125284dfd712c/lasheightCl.las"

LAStools console output

The system cannot find the path specified. 

Results: {'': None}

Results: {'classifiedLAZ': 'C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_avyDJN/64a1ce455e2f484c985125284dfd712c/lasheightCl.las'}

WhiteboxTools command:

"C:/OSGEO4W64/WBT/whitebox_tools.exe" -v --run=LidarPointDensity --input="C:\Users\ewynter.CAMSOL\AppData\Local\Temp\processing_avyDJN\64a1ce455e2f484c985125284dfd712c\lasheightCl.las" --returns="all" --resolution=2.0 --radius=10.0 --exclude_cls="0,1,3,4,5,7,8" --output="C:\Users\ewynter.CAMSOL\AppData\Local\Temp\processing_avyDJN\6e30f7353a93496181adf8d2afc242ec\gpd.tif"

WhiteboxTools output:

********************************
* Welcome to LidarPointDensity *
* Powered by WhiteboxTools *
* www.whiteboxgeo.com <http://www.whiteboxgeo.com>  *
********************************

thread 'main' panicked at 'The system cannot find the file specified. (os error 2)', whitebox-tools-app\src\main.rs:72:21
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

Process returned error code 101

Results: {'output': 'C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_avyDJN/6e30f7353a93496181adf8d2afc242ec/gpd.tif'}

Unable to execute algorithm
Could not load source layer for input: C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_avyDJN/6e30f7353a93496181adf8d2afc242ec/gpd.tif not found

Unable to execute algorithm
Could not load source layer for input: C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_avyDJN/6e30f7353a93496181adf8d2afc242ec/gpd.tif not found

Execution failed after 1.35 seconds

 

Loading resulting layers

Algorithm '0NE (One-Step-Processing)' finished

**Comment by ericawynter** [comment]
Hi Stefan,

 

I just tried to run the liidarpointdensity tool in Whitebox directly and have come across what may be the cause. The left screen below shows QGIS lidarpoint density tool box.  As you can see it shows my folder as empty.  If you scroll right you can see my filemanager and all my laz files are in this folder.  I am not sure why this is happening.

 

Regards

 

Erica

**Comment by stefaneichert** [comment]
Hi Erica!
thank you for reporting the errors. 
To exclude other reasons for the error I have a few questions:

Did you install only the plugins via QGIS or did you also install the Lastools and whitebox software on your computer?
Is there a folder C:\lastools on your machine with the required lastools files?
Also there should be a folder C:/OSGEO4W64/WBT/ with the whitebox executables
Are they there?

What operating system are you using?

Thanks and best regards,
Stefan

**Comment by ericawynter** [comment]
Hi Stefan,

 

Apologies I realised last night that I had tried to run a laz file.  I had
deleted my las files because of an issue we had with google drive corrupting
the files, (long story but we finally narrowed it down to a google drive
file transfer issue)in any case I now have the correctly projected files in
laz and will convert them today and retry once we have power restored.

 

Regards

 

Erica

**Comment by ericawynter** [comment]
Hi Stefan,

 

 

I have taken screen shots all along so you can see, if you would like the
files I am working with I can WeTransfer to you, although I have no issues
in global mapper, QGIS or golden surfer opening them.   The message keeps
changing slightly every time I run the process but what remains constant is
the missing
:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_kLYWCd/03826f8112fe4b1c
b58fdc0678e62fdc/gpd.tif not found

 

All I can assume is that some integral link from bruy is still missing and I
honestly am not a programmer and would not know how to resolve this.

 

Regards

 

Erica

 

 



Error 

QGIS version: 3.22.3-Białowieża

QGIS code revision: 1628765ec7

Qt version: 5.15.2

Python version: 3.9.5

GDAL version: 3.4.1

GEOS version: 3.10.0-CAPI-1.16.0

PROJ version: Rel. 8.2.1, January 1st, 2022

PDAL version: 2.3.0 (git-version: 9f35b7)

Algorithm started at: 2022-03-12T16:20:39

Algorithm '0NE (One-Step-Processing)' starting.

Input parameters:

{ 'CRS' : QgsCoordinateReferenceSystem('EPSG:32735'), 'GPD' : True,
'InputFilelaslaz' : 'H:\\PHDATA\\PHLas\\Full _048.las', 'LAS' :
'F:/PH_OneStep/OpenL_048.laz', 'LVD' : True, 'LowNoise' : False,
'SetCellSize' : 0.5, 'VisualisationCM' : True, 'VisualisationDFM' : True,
'VisualisationDfME' : True, 'VisualisationHS' : True, 'VisualisationOPN' :
True, 'VisualisationSVF' : True, 'VisualisationVAT' : True, 'classLas' :
False, 'prefix' : '' }

 

LAStools command line

C:\LAStools\bin\lasground -v -i "H:\PHDATA\PHLas\Full _048.las"
-ignore_class 7 -city -ultra_fine -o
"C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_kLYWCd/17d2af80054644
379898aac1a4f741e9/lasground1.laz"

LAStools console output

The system cannot find the path specified. 

Results: {'': None}

LAStools command line

C:\LAStools\bin\lasheight -v -i
"C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_kLYWCd/17d2af80054644
379898aac1a4f741e9/lasground1.laz" -o
"C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_kLYWCd/58723606853e4d
e78ccd0c520ecc4b75/lasheight.laz" -ignore_class 0 1 3 4 5 6 7 8 9 10 11 12

LAStools console output

The system cannot find the path specified. 

Results: {'': None}

LAStools command line

C:\LAStools\bin\lasclassify -v -i
"C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_kLYWCd/58723606853e4d
e78ccd0c520ecc4b75/lasheight.laz" -o
"C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_kLYWCd/2237907f3d2b4d
98a57be35a43253012/lasclassify.laz" -small_trees

LAStools console output

The system cannot find the path specified. 

Results: {'': None}

LAStools command line

C:\LAStools\bin\lasground -v -i
"C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_kLYWCd/2237907f3d2b4d
98a57be35a43253012/lasclassify.laz" -wilderness -ultra_fine -o
"C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_kLYWCd/0e51f09b315e4f
0ca8e34680275530c0/lasground2.laz" -ignore_class 6 7

LAStools console output

The system cannot find the path specified. 

Results: {'': None}

LAStools command line

C:\LAStools\bin\lasheight -v -i
"C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_kLYWCd/0e51f09b315e4f
0ca8e34680275530c0/lasground2.laz" -ignore_class 6 -ignore_class 7
-classify_below -0.25 7 -classify_between -0.2 0.2 2 -classify_between 0.5
2.0 3 -classify_above 2.0 5 -o
"C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_kLYWCd/b313e64a03e047
e4be49d42ba457db15/lasheightCl.las"

LAStools console output

The system cannot find the path specified. 

Results: {'': None}

Results: {'classifiedLAZ':
'C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_kLYWCd/b313e64a03e047
e4be49d42ba457db15/lasheightCl.las'}

LAStools command line

C:\LAStools\bin\laszip -i
"C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_kLYWCd/b313e64a03e047
e4be49d42ba457db15/lasheightCl.las" -o "F:/PH_OneStep/OpenL_048.laz"

LAStools console output

The system cannot find the path specified. 

Results: {'': None}

WhiteboxTools command:

"C:/OSGEO4W64/WBT/whitebox_tools.exe" -v --run=LidarPointDensity
--input="C:\Users\ewynter.CAMSOL\AppData\Local\Temp\processing_kLYWCd\b313e6
4a03e047e4be49d42ba457db15\lasheightCl.las" --returns="all" --resolution=2.0
--radius=10.0 --exclude_cls="0,1,3,4,5,7,8"
--output="C:\Users\ewynter.CAMSOL\AppData\Local\Temp\processing_kLYWCd\03826
f8112fe4b1cb58fdc0678e62fdc\gpd.tif"

WhiteboxTools output:

********************************
* Welcome to LidarPointDensity *
* Powered by WhiteboxTools *
* www.whiteboxgeo.com <http://www.whiteboxgeo.com>  *
********************************

thread 'main' panicked at 'The system cannot find the file specified. (os
error 2)', whitebox-tools-app\src\main.rs:72:21
note: run with `RUST_BACKTRACE=1` environment variable to display a
backtrace

Process returned error code 101

Results: {'output':
'C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_kLYWCd/03826f8112fe4b
1cb58fdc0678e62fdc/gpd.tif'}

Unable to execute algorithm
Could not load source layer for input:
C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_kLYWCd/03826f8112fe4b1
cb58fdc0678e62fdc/gpd.tif not found

Unable to execute algorithm
Could not load source layer for input:
C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_kLYWCd/03826f8112fe4b1
cb58fdc0678e62fdc/gpd.tif not found

Execution failed after 0.62 seconds

 

Loading resulting layers

Algorithm '0NE (One-Step-Processing)' finished

QGIS version: 3.22.3-Białowieża

QGIS code revision: 1628765ec7

Qt version: 5.15.2

Python version: 3.9.5

GDAL version: 3.4.1

GEOS version: 3.10.0-CAPI-1.16.0

PROJ version: Rel. 8.2.1, January 1st, 2022

PDAL version: 2.3.0 (git-version: 9f35b7)

Algorithm started at: 2022-03-12T16:23:18

Algorithm '0NE (One-Step-Processing)' starting.

Input parameters:

{ 'CRS' : QgsCoordinateReferenceSystem('EPSG:32735'), 'GPD' : True,
'InputFilelaslaz' : 'H:\\PHDATA\\PHLas\\Full _048.las', 'LAS' :
'F:/PH_OneStep/OpenL_048.laz', 'LVD' : True, 'LowNoise' : False,
'SetCellSize' : 0.5, 'VisualisationCM' : True, 'VisualisationDFM' : True,
'VisualisationDfME' : True, 'VisualisationHS' : True, 'VisualisationOPN' :
True, 'VisualisationSVF' : True, 'VisualisationVAT' : True, 'classLas' :
False, 'prefix' : '' }

 

LAStools command line

C:\LAStools\bin\lasground -v -i "H:\PHDATA\PHLas\Full _048.las"
-ignore_class 7 -city -ultra_fine -o
"C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_kLYWCd/fda66c67476049
89ace50e97ccb4f60b/lasground1.laz"

LAStools console output

The system cannot find the path specified. 

Results: {'': None}

LAStools command line

C:\LAStools\bin\lasheight -v -i
"C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_kLYWCd/fda66c67476049
89ace50e97ccb4f60b/lasground1.laz" -o
"C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_kLYWCd/a4be83a0baab47
13a372ea5a72584e70/lasheight.laz" -ignore_class 0 1 3 4 5 6 7 8 9 10 11 12

LAStools console output

The system cannot find the path specified. 

Results: {'': None}

LAStools command line

C:\LAStools\bin\lasclassify -v -i
"C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_kLYWCd/a4be83a0baab47
13a372ea5a72584e70/lasheight.laz" -o
"C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_kLYWCd/9b77f40ddb5643
aa910acbc3481b3359/lasclassify.laz" -small_trees

LAStools console output

The system cannot find the path specified. 

Results: {'': None}

LAStools command line

C:\LAStools\bin\lasground -v -i
"C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_kLYWCd/9b77f40ddb5643
aa910acbc3481b3359/lasclassify.laz" -wilderness -ultra_fine -o
"C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_kLYWCd/dc7bfd13a93f4b
8a8eb4d6c72b6675d5/lasground2.laz" -ignore_class 6 7

LAStools console output

The system cannot find the path specified. 

Results: {'': None}

LAStools command line

C:\LAStools\bin\lasheight -v -i
"C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_kLYWCd/dc7bfd13a93f4b
8a8eb4d6c72b6675d5/lasground2.laz" -ignore_class 6 -ignore_class 7
-classify_below -0.25 7 -classify_between -0.2 0.2 2 -classify_between 0.5
2.0 3 -classify_above 2.0 5 -o
"C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_kLYWCd/d2d2fc0c50684c
3fb3177280ae4061b1/lasheightCl.las"

LAStools console output

The system cannot find the path specified. 

Results: {'': None}

Results: {'classifiedLAZ':
'C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_kLYWCd/d2d2fc0c50684c
3fb3177280ae4061b1/lasheightCl.las'}

LAStools command line

C:\LAStools\bin\laszip -i
"C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_kLYWCd/d2d2fc0c50684c
3fb3177280ae4061b1/lasheightCl.las" -o "F:/PH_OneStep/OpenL_048.laz"

LAStools console output

The system cannot find the path specified. 

Results: {'': None}

WhiteboxTools command:

"C:/OSGEO4W64/WBT/whitebox_tools.exe" -v --run=LidarPointDensity
--input="C:\Users\ewynter.CAMSOL\AppData\Local\Temp\processing_kLYWCd\d2d2fc
0c50684c3fb3177280ae4061b1\lasheightCl.las" --returns="all" --resolution=2.0
--radius=10.0 --exclude_cls="0,1,3,4,5,7,8"
--output="C:\Users\ewynter.CAMSOL\AppData\Local\Temp\processing_kLYWCd\9e5f4
1d3d3574047854ff32811245c24\gpd.tif"

WhiteboxTools output:

********************************
* Welcome to LidarPointDensity *
* Powered by WhiteboxTools *
* www.whiteboxgeo.com <http://www.whiteboxgeo.com>  *
********************************

thread 'main' panicked at 'The system cannot find the file specified. (os
error 2)', whitebox-tools-app\src\main.rs:72:21
note: run with `RUST_BACKTRACE=1` environment variable to display a
backtrace

Process returned error code 101

Results: {'output':
'C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_kLYWCd/9e5f41d3d35740
47854ff32811245c24/gpd.tif'}

Unable to execute algorithm
Could not load source layer for input:
C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_kLYWCd/9e5f41d3d357404
7854ff32811245c24/gpd.tif not found

Unable to execute algorithm
Could not load source layer for input:
C:/Users/ewynter.CAMSOL/AppData/Local/Temp/processing_kLYWCd/9e5f41d3d357404
7854ff32811245c24/gpd.tif not found

Execution failed after 0.60 seconds

 

Loading resulting layers

Algorithm '0NE (One-Step-Processing)' finished

**Comment by stefaneichert** [comment]
Do Lastools and Whitebox actually work correctly on your system if you try to run an algorithm directly from the processing toolbox?

**Comment by ericawynter** [comment]
Hi Stefan,

 

I am actually in the process of reloading everything from scratch on my laptop.  As soon as it is done I will try again.  Any advice before I start for order of install and which version of QGIS would be appreciated.  Apologies I have been so busy with my research and work that I have fallen behind on getting my laptop sorted for TRAIL.

 

Regards

 

Erica

**Comment by ericawynter** [comment]
Hi Stefan,

 

I am going to factory reset my laptop this weekend and start from scratch with loading QGIS an the OpenLidarToolbox.

 

Please can you check which versions of QGIS you are running the Open Lidar Toolbox on.  I would like to install the QGIS version/s most likely to work.  

 

Regards

 

Erica

**Comment by stefaneichert** [comment]
Dear Erica,
The plugin available from the official QGIS plugin repository was tested on QGIS 3.1x
The development version was also already tested on QGIS Tisler 3.2.4. It is not yet available via the built in plugin installation of  QGIS. You can install it from .zip. The file can be downloaded here: https://github.com/stefaneichert/OpenLidarToolbox/raw/develop/src/QGISplugin/open_lidar_toolbox.zip

Could you please check if LasTools and Whitebox run within your QGIS distribution separately from OpenLiDARToolbox, so we can see if the problem is with our Toolbox or has some other reasons.

Best,
Stefan

**Comment by ericawynter** [comment]
Hi Stefan,

 

Just to bring you up to dat, I loaded using the Tisler as you suggested.  I
am still having problems with the lastools and whitebox tools functions.  I
have followed whitebox tools' new instructions for the QGIS plugin as per
their website and it still will not run the functions.  Will try on QGIS 3.1
and let you know but at this rate I will be attending with a non functioning
QGIS as I leave South africa on the 21st.  

 

Regards

 

Erica

**Comment by stefaneichert** [comment]
The Error Message submitted is caused if the additional Whitebox resp. LAStools files are not installed. Please refer to the updated installing instructions: 
https://github.com/stefaneichert/OpenLidarToolbox#installation
or to the troubleshooting section:
https://github.com/stefaneichert/OpenLidarToolbox#troubleshooting

**Comment by ericawynter** [comment]
Hi Stefan,

 

I managed to get it working today.  Apologies for the delay, I have been
crazy busy with training on our software and have only got some relief today
to revisit the QGIS plugins.  I had reloaded the whiteboxtools last week but
the lastools was still throwing an error which naturally caused an error
down the line.  The last tweak was an easy one, I had forgotten to change
the link in QGIS to refer to the folder lastools was stored in.  The problem
of repeating steps when doing installations over and over is that you start
thinking you had done something but have omitted it in the most recent
iteration.

 

Thank you for your patience, the small test I ran this morning looks
promising and I look forward to learning more in a few weeks time.

 

Regards 

 

Erica Wynter
